### ISSUE TITLE
Wrong P-value for Fisher test with mid-P correction

### ISSUE THREAD

**Issue by huftis** [ocbe-uio/contingencytables#51]
The Fisher mid-*P* calculations seems to be incorrect for *some* tables. For the following table, I get a *P*-value of 1:

``` r
library(contingencytables)
#> Welcome to contingencytables 2.1.1.
#> Please run ?chapX or list_functions(x) to see the functions related to chapter X.
x = matrix(c(745, 127, 3, 2), ncol = 2)
Fisher_midP_test_2x2(x)
#> The Fisher mid-P test (Fisher-Irwin): P = 1.00000
```

However, the *P*-value for a Fisher test *without* mid-*P* correction is only 0.159, so the *P*-value *with* the correction should be lower:

```r
Fisher_exact_test_2x2(x)
#> The Fisher exact test (Pearson): P = 0.15903
fisher.test(x)$p.value
#> [1] 0.1590269
```

For comparison, the coin package reports a *P*-value of 0.09.

```r
library(coin)
midpvalue(independence_test(as.table(x), distribution="exact"))
#> [1] 0.09193139
````

(The coin package reports the *same* *P*-values as `Fisher_midP_test_2x2()` does for the four datasets listed on the `?Fisher_midP_test_2x2` help page.)

<sup>Created on 2023-10-04 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by wleoncio** [comment]
Thanks for reporting, we can reproduce the issue and will fix it ASAP.

**Comment by wleoncio** [comment]
So the difference between the function examples and the one provided in the OP is that the latter triggers the `else` condition in the chunk below:

https://github.com/ocbe-uio/contingencytables/blob/0c299a7554a034576d26f4ef4682fa0d90bd0215/R/Fisher_midP_test_2x2.R#L31-L42

So basically any matrix with a large enough value in the first cell would yield a p-value of 1.  I'll study the methodology a bit better and get back ASAP.

**Comment by huftis** [comment]
And if we exchange the order of the rows and columns (to avoid triggering the `else` branch), we get the same results as with the coin package:

```r
Fisher_midP_test_2x2(x[2:1, 2:1])
# The Fisher mid-P test (Fisher-Irwin): P = 0.09193
```

**Comment by wleoncio** [comment]
Hi again,

Looks like the issue was caused by us using an old algorithm to calculate `Tobs`. We went back to the last version of the original (MATLAB) code and fixed it on commit 708d8df44e13cdec91102a7eae1503a6604f5268.

There should be no need to swap rows.

I'll submit a new version containing this fix on CRAN today, thanks again for the contribution!
